### DISCUSSION TITLE
An alternative inpainting model that supports inpainting, removal and outpainting. It can be can be used with controlnet

### DISCUSSION THREAD

**Discussion by zengyh1900** [Mikubill/sd-webui-controlnet#2323] [Show and tell]
Hi all,

I want to share our recent model for image inpainting, PowerPaint. 
It can be the most powerful inpainting model that enables text-guided object inpainting, text-free object removal, and image outpainting, as an open-source alternative of Adobe Firefly. 

- youtube: [https://youtube.com/watch?v=7QsiY1JQUfg](https://t.co/1hQYqEcXqw)
- online demo: https://openxlab.org.cn/apps/detail/rangoliu/PowerPaint
- arxiv:  [https://arxiv.org/abs/2312.03594](https://t.co/DAnxCG0V0a)
- code: https://github.com/open-mmlab/mmagic/tree/main/projects/powerpaint

We are looking forward to any feedback!

**Comment by zengyh1900** [comment]
Notably, PowerPaint can be used with ControlNet (e.g., canny edge, pose, etc.)!

**Comment by zengyh1900** [comment]
Compared with Adobe Firefly,
![20231212-150321](https://github.com/Mikubill/sd-webui-controlnet/assets/7473216/0e69f929-bf72-4d0d-9928-1c58c70d17a3)

**Comment by zengyh1900** [comment]
Some more results here. Notably, all these are achieved by using only a single model without tailored workflow!
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/7473216/046d0bea-23fc-40b5-954e-abff91bc6c79)

**Comment by huchenlei** [comment]
Reference discussion: https://github.com/Acly/krita-ai-diffusion/issues/221

**Reply by lllyasviel** [reply]
is the model XL or 1.5?

**Reply by huchenlei** [reply]
> Our method does have two inherent limitations. Firstly,
the synthesis quality can be constrained by the capabilities
of the underlying text-to-image diffusion model. We plan to
alleviate this limitation by extending our method to a more
powerful text-to-image diffusion model, such as SDXL [22]

Looks like SD1.5.

**Comment by sdbds** [comment]
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/8085926/3ca2d5e3-ac7b-4921-ae0e-80be9dc56ef3)
I think they use diffusers key

**Comment by sdbds** [comment]
i transform it into sd model,now we can use it in fooocus、comfyui and webui
https://huggingface.co/bdsqlsz/PowerPaint-V1-stable-diffusion-inpainting

**Reply by lllyasviel** [reply]
does it have designed layers or all keys exactly same as 1.5 inpaint model?

**Reply by sdbds** [reply]
same with 1.5 inpainting.

**Reply by sdbds** [reply]
https://gist.github.com/sdbds/cf3f66635ae1d03a903be5f2d29ef564

**Reply by Woisek** [reply]
And how is it now used? 🤔

**Reply by sdbds** [reply]
use as stable diffusion inpainting model,but their shaped-guide use fitting degree not implement in webui now

**Reply by Woisek** [reply]
Does this mean, it's still not usable ... ? And how about the Photoshop Plugins? Do they require something special to use this as inpainting model?

**Reply by sdbds** [reply]
They actually had one way to use shaped guide and extra arg called fitting degree to use.
But you can use normal way such as text inpaint、remove something and outpainting.

**Reply by Woisek** [reply]
That sounds cool then. Will have a look. Thanks! 👍

**Comment by dilectiogames** [comment]
can someone answer like I'm 5years old if this can be used with my local automatic installation and what I need to do to make this work.

do I need to download a model for controlnet or what do I need to do?

**Comment by zhuang2002** [comment]
PowerPaint Update

We've released an updated model with enhanced stability, now available for interactive testing on our online demo.

Paper: https://arxiv.org/abs/2312.03594
HomePage: https://powerpaint.github.io/
Online Demo: https://openxlab.org.cn/apps/detail/rangoliu/PowerPaint
Code: https://github.com/open-mmlab/mmagic/tree/main/projects/powerpaint

**Reply by Woisek** [reply]
@sdbds Would you again be so kind and convert this for us, please? 🙂
@zhuang2002 It would be a good idea, to (also) release a converted safetensor file for the community for those who are not able to convert this themselves. 👍 Then we all could appreciate this nice work even more.

**Reply by sdbds** [reply]
updated now
https://huggingface.co/bdsqlsz/PowerPaint-V1-stable-diffusion-inpainting/

**Reply by Woisek** [reply]
Many thanks! 👍

**Reply by Fishkin** [reply]
Can't get this to work. I mean, it "works" except it generates completely unrelated stuff. I just use it as model and inpaint normally, do I have to do something extra? ControlNet inpainting model on top of it doesn't help much either.

**Reply by soldivelot** [reply]
i made it work my merging with another base model
![图片](https://github.com/Mikubill/sd-webui-controlnet/assets/84898012/dce7a61d-7143-4bc7-ab8e-9cbacf4269f9)

**Comment by IndigoDosSantos** [comment]
Shape-guided object inpainting is not yet supported in webui, or is it?

**Reply by huchenlei** [reply]
You can use ControlNet during inpainting. I am not sure what do you mean "shape"? (Depth? Canny?)

**Reply by IndigoDosSantos** [reply]
PowerPaint was trained on three task prompts:

- P<sub>obj</sub> for text-guided object inpainting
- P<sub>ctxt</sub> for context-aware image inpainting
- P<sub>shape</sub> for shape-guided inpainting (If I've understood it correctly...)

In the gradio demo of PowerPaint there is a slider that controls how closely the inpainted object aligns with the shape of the mask and supposedly works without any additional ControlNet conditions.

**Comment by alexgilseg** [comment]
Is this usable with A1111?

**Reply by IndigoDosSantos** [reply]
Yes, it is.

**Comment by zhuang2002** [comment]
We have recently developed a new version of [PowerPaint](https://github.com/zhuang2002/PowerPaint), drawing inspiration from Brushnet. The [Online Demo](https://openxlab.org.cn/apps/detail/rangoliu/PowerPaint) has been updated accordingly. In the coming days, we intend to make the model weights available as open source.

The PowerPaint model possesses the ability to carry out diverse inpainting tasks, such as object insertion, object removal, shape-guided object insertion, and outpainting. Notably, we have retained the cross-attention layer that BrushNet had removed, which is essential for task prompt input.

|  | Object insertion | Object Removal|Shape-guided Object Insertion|Outpainting|
|-----------------|-----------------|-----------------|-----------------|-----------------|
| Original Image| ![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/bf91a1e8-8eaf-4be6-b47d-b8e43c9d182a)|![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/c7e56119-aa57-4761-b6aa-56f8a0b72456)|![image](https://github.com/Sanster/IOPaint/assets/108931120/cbbfe84e-2bf1-425b-8349-f7874f2e978c)|![cropinput](https://github.com/Sanster/IOPaint/assets/108931120/134bb707-0fe5-4d22-a0ca-d440fa521365)|
| Output| ![image](https://github.com/Sanster/IOPaint/assets/108931120/ee777506-d336-4275-94f6-31abf9521866)| ![image](https://github.com/Sanster/IOPaint/assets/108931120/e9d8cf6c-13b8-443c-b327-6f27da54cda6)|![image](https://github.com/Sanster/IOPaint/assets/108931120/cc3008c9-37dd-4d98-ad43-58f67be872dc)|![image](https://github.com/Sanster/IOPaint/assets/108931120/18d8ca23-e6d7-4680-977f-e66341312476)|
